docker/auth.go

```diff
@@ -167,6 +177,17 @@ func (rid ImageRepoDetails) String() string {
 	return string(d)
 }
 
+// Print returns a string for logging purpose.
+func (rid ImageRepoDetails) Print() string {
```

Comment:
Should this be String() - ie for safety, the default behaviour should be to mask the tokens.
Then if people use "%s" they can't accidentally leak data.
If we need a method to print out the tokens, then that should be the exception, not the usual behaviour.

Reply:
Yeah, agreed. The usual behaviour be masking the tokens.